### DISCUSSION TITLE
Submissions from Others

### DISCUSSION THREAD

**Discussion by ArchEzekiel** [malcolmriley/unused-renders#4] [General]
What are your thoughts on other people making additions to the Repo? Ive been wanting practice and I think this is a perfect opportunity to give back to the community, so its a win-win

**Comment by malcolmriley** [comment]
Hello!

Thank you for reaching out - I'm very open to the idea, in fact I think it is fantastic. I would, however, want to make sure that contributors receive fair treatment. My concerns are reducible to three broad categories:

1. **Licensing:** The works in the repository are all currently [CC-BY-4.0](https://creativecommons.org/licenses/by/4.0/), but I would want any other contributing artist to be able to select a license that suits them. The GitHub repository main page has only a single spot for declaring licenses.
2. **Adequate Credit:** The works in this repository are all nominally under my name (as this is a personal GitHub account and not an organization account). Since I am currently the only contributor, this is okay, but if there were other contributors I would want to ensure  that they receive attribution for their work.
3. **Provenance and Ownership:** Once again, since this repository currently contains only original works by me, I can freely offer the works contained within. I would never want to be in the situation of accidentally offering the work of another artist in a context they wouldn't approve of themselves (and I have no real way of predicting what that would be). My commits are all signed, and I think I would ask contributors to *this* repository to do the same in order to keep record of the provenance of the works contained within.

This isn't to say that I don't want to accept other contributors, I'm just not quite sure how to give them a fair deal with the way that this repository is currently set up. For instance, maybe a note in the `README` or something is enough for some, but I wouldn't want to turn away a contributor who wouldn't be satisfied with that.

If you (or anyone else reading) has any insights to share, please do!

**Reply by ArchEzekiel** [reply]
In response to your points in order:

1. I think that if someone would like to contribute here, they should accept using the current license your already using, and that if they feel strongly about using a different license, its their prerogative to start their own repository. Of course, as it is your personal repository there would be some risk of you changing the license for other contributors' work, potentially without their consent (if they don't respond to communication for example) so it may be worth adding a note somewhere making a point clear about whether or not that is on the table. If you were dead-set on letting artists submit under different licenses, than you could mark a folder for each license and have submissions sorted by license.
2. I can absolutely understand the worry about credit attribution, and this technically could be solved with sorted folders as well... however this could lead to some bloating if there are a large variety of submissions. Of course, if all submissions are done through PRs, then the GitHub user would be attached to each commit and that would act as a signature. Sadly, there will be some confusion for attribution if this continues to be a personal repository rather than an organization, since checking the signature probably wouldn't be done by everyone who uses this work, and some people might incorrectly attribute others works as yours. I would love for this to become a central repository so making it an organization might solve a few of these problems, and I think it would be wonderful to have one place for new modders to pull from rather than feel overwhelmed by the process of making their own art to start.
3. This is a mix of both of the previous points from what I understand, and perhaps if you wanted to fully open up to other contributions, maybe something like a Terms of Service which would basically inform contributors that if they submit something here that they are agreeing that all their works are free of use everywhere. The best way to prevent confusion of what is done with work is to be as open and honest as possible. Make it clear if that someone doesn't understand they can ask questions, because being open is the goal.

**Reply by malcolmriley** [reply]
Thank you for your insights!

It's possible I'm overthinking this. Clearly, if somebody is approaching me about adding work to the repository, they're satisfied with the way it works as-is, so thank you also for humoring me.

I like the idea of subfolders; perhaps per-contributor? That way each could add a license and `README` with contact information if they choose. I very much want to preserve the option for a prospective mudpack author to easily reach out to the original artist to make requests or commission.

As far as provenance goes, I think you're right that the commit history is a good-enough proof of the point of origin of each asset, with commit signing being an optional additional layer of surety/security. I've been digitally "signing" my work by embedding authorship information into the metadata - I'll make a note to create a tutorial for how to do this.

**Comment by malcolmriley** [comment]
I also wanted to ask, since you said you were practicing - have you taken a look at the [guide wiki](https://github.com/malcolmriley/unused-renders/wiki) section I've been developing? Being that developing artwork for Factorio is a bit more complicated than for, say, Minecraft, this is another good opportunity for community resource-sharing.

I'd also be interested in feedback regarding the tutorial series (whether it is any good) or any insights into broadening it into a community knowledge-sharing effort.

**Reply by ArchEzekiel** [reply]
I've skimmed it and I plan to chug through it over the next week or so, Ill take any written resource I can get because I struggle sometimes with the pacing of video tutorials. I can write down notes and thoughts as I do so, and I'll probably start a new discussion thread for that feedback once I have it. Factorio is definitely a different beast than Minecraft that is for sure, both have their quirks though

**Reply by malcolmriley** [reply]
I haven't received much feedback on the guide series so far, so if it's not too much trouble I would be very grateful for it! (This goes for you or any other eavesdroppers.)

**Comment by malcolmriley** [comment]
With some consideration I've decided on the one-subdirectory-per-contributor idea (thank you for the suggestion, @ArchEzekiel!). I'm still open to feedback on this, but I believe the following policy to be "good enough" for the foreseeable future:

- Each individual contributor will maintain a named subdirectory within the repository
- Individual contributors will optionally provide a `README`, with the suggestion that they provide a brief bio and contact information so as to potentially receive future requests and/or commissions
- Individual contributors may provide an overriding license within their subdirectory, in the case that they don't wish to use the top-level license. Licenses within the subdirectory should be taken to supersede (and replace) the top-level license `CC-BY-4.0`.
- A strong recommendation (but not obligation) to use commit signing and authorship metadata embeds so as to guarantee provenance of the assets within the repository

I'm away from my primary workstation at the moment, but when I return in a week or so I'll reorganize this repository to accommodate those changes (PR templates, update root `README` etc).

**Reply by ThelianTech** [reply]
Just reading through this, I would definitely agree to all of what has been discussed. and If you want to be the sole maintainer of the repository (by accepting PR's etc) then this is perfect. however I would suggest if you would want a few solid team members to help maintain this repository and be able to have multiple people have access for PR submissions and acceptance, it would probably be better to clone this repository under a github organization after the repository restructure.

**Reply by malcolmriley** [reply]
Thank you for your insights! I'm not opposed to turning this endeavor into a team effort - I just don't know if there's enough work to go around, so to speak. I have yet to receive any PRs for asset contributions (the only murmurings of such things are those you see here) and so I must admit I've put the reorganization of this project a bit on the back burner.

I do wonder if with the imminence of Space Age this is just a slow period for the modding community - I know that I'm waiting for the release to resume work on my in-progress Factorio mod projects, so I imagine other creators have made similar decisions.

**Reply by ThelianTech** [reply]
I can agree there. most Modding seems to be slowing down now since the released of the information for Space Age's Release date in October. I assume that it will pick back up around then as older mods/modders update for the 2.0 free optimizations but also for expanding on integration with space age DLC. I too mostly took a break from working on my overhaul mod for almost 2 months since I've been busy doing other things as well as wanting to mostly wait till after 2.0 to release an alpha release of my Overhaul mod.

**Reply by malcolmriley** [reply]
I'll watch for an uptick towards the end of November then - I'm assuming we'll all be playing Space Age instead of modding throughout October!
